torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -699,16 +690,33 @@ void TensorExprKernel::bindInput(const torch::jit::Value* input) {
   auto const& t = input->type();
   switch (t->kind()) {
     case TypeKind::TensorType: {
-      Buffer in_buffer = texprBuffer(input);
+      auto tt = input->type()->cast<TensorType>();
```

Comment:
I think we definitely don't want to do this for contiguous tensors. It makes further optimizations more difficult, unless all strides are also statically known as well. 

Does the profiling executor tell us whether this kernel targets contiguous tensors or not? Thanks!

Reply:
Ooh.  I did not realize this, but the profiling executor also profiles strides as well, already.  Woah.  I can definitely make use of that.